cpp-package/example/utils.h

```diff
@@ -27,6 +27,18 @@
 
 using namespace mxnet::cpp;
 
+#define TRY \
+  try {
+#define CATCH \
+  } catch(dmlc::Error &err) { \
+    LG << "Status: FAIL";\
+    LG << "With Error: " << MXGetLastError(); \
+    return 1; \
+  } catch(...) { \
```

Comment:
i think you should only have dmlc::Error catch and remove the catch all. If there is any other exception thrown it could be because of some MXNet backend showing non dmlc::Errors (which is very unlikely) or some other exception thrown from the example code because of user or programming error. Users should be able to see what these exceptions are without having to modify utils.